### PR TITLE
fix: refresh user list after removal (AR-2477)

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2477" title="AR-2477" target="_blank">StoryAR-2477</a>  Handle incremental events for teams
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Refreshing search users list after one of team member was removed or left a team.

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4618146)(4618146)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4618146)(4618146)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt [hunk 1]
```diff
@@ -208,20 +206,17 @@ abstract class SearchPeopleViewModel(
...
-            }
-            is SearchResult.Success -> {
-                emit(
+                }
+                is SearchResult.Success -> {
                     SearchResultState.Success(result.contacts)
-                )
+                }
             }
         }
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt]: I do not think there is need for a change to a function, you can keep 

protected val initialContactResultFlow as the rest as a variable, and keep the rest you did with a map { }

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt [hunk 2]
```diff
@@ -84,11 +79,14 @@ open class SearchAllPeopleViewModel(
...
-            is GetAllContactsResult.Failure -> SearchResult.Failure(R.string.label_general_error)
-            is GetAllContactsResult.Success -> SearchResult.Success(result.allContacts.map(contactMapper::fromOtherUser))
+    override suspend fun getInitialContacts(): Flow<SearchResult> = getAllKnownUsers()
+        .map { result ->
+            when (result) {
+                is GetAllContactsResult.Failure -> SearchResult.Failure(R.string.label_general_error)
+                is GetAllContactsResult.Success -> SearchResult.Success(result.allContacts.map(contactMapper::fromOtherUser))
+            }
         }
+        .flowOn(dispatcher.io())
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt]: I do not think there is need for a change to a function, you can keep 

protected val initialContactResultFlow as the rest as a variable, and keep the rest you did with a map { }

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4642217)(4642217)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4642217)(4642217)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt]: I fixed issue with using suspend fun for flowables

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt]: Same here

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/5/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt [hunk 3]
```diff
@@ -275,7 +269,7 @@ abstract class SearchPeopleViewModel(
         }
     }
 
-    abstract suspend fun getInitialContacts(): SearchResult
+    abstract fun getInitialContacts(): Flow<SearchResult>
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchPeopleViewModel.kt:271]: Nice catch, good topic for discussion on the AC. I'll prepare a couple a detekt rules for this if we agree on this topic.

**Review by yamilmedina — APPROVED** [review]
Excellent job as always !

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1084/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4652621)(4652621)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4652621)(4652621)-dev-release.apk)

**Pull request merged**